src/main/java/io/cryostat/agent/WebServer.java

```diff
@@ -164,6 +171,59 @@ public String description() {
         }
     }
 
+    private class CompressionFilter extends Filter {
+
+        @Override
+        public void doFilter(HttpExchange exchange, Chain chain) throws IOException {
+            List<String> encodings = new ArrayList<>();
+            List<String> rawEncodings =
+                    exchange.getRequestHeaders().getOrDefault("Accept-Encoding", List.of());
+            rawEncodings.forEach(
+                    raw -> {
+                        if (raw.contains(",")) {
+                            encodings.addAll(Arrays.asList(raw.replaceAll("\\s", "").split(",")));
+                        } else {
+                            encodings.add(raw);
+                        }
+                    });
+            String actualEncoding = null;
+            priority:
+            for (String requestedEncoding : encodings) {
+                switch (requestedEncoding) {
+                    case "deflate":
+                        actualEncoding = requestedEncoding;
+                        exchange.setStreams(
+                                exchange.getRequestBody(),
+                                new DeflaterOutputStream(exchange.getResponseBody()));
+                        break priority;
+                        // TODO gzip encoding breaks communication with the server, need to
+                        // determine why and re-enable this
+                        // case "gzip":
+                        // actualEncoding = requestedEncoding;
+                        // exchange.setStreams(
+                        //         exchange.getRequestBody(),
+                        //         new GZIPOutputStream(exchange.getResponseBody()));
+                        // break priority;
+                    default:
+                        break;
+                }
+            }
+            Optional.ofNullable(actualEncoding)
```

Comment:
Is there a reason other than style to use `Optional.ofNullable` + `ifPresentOrElse` vs `if(actualEncoding == null)` + `else` ?

Reply:
No, just a style thing (or bad cleanup after some refactoring).